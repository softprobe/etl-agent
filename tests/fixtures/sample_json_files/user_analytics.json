[
  {
    "user_id": "user_123",
    "session_id": "sess_abc789",
    "timestamp": "2024-01-15T09:45:30.123Z",
    "event_type": "page_view",
    "page": "/dashboard",
    "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36",
    "ip_address": "192.168.1.100",
    "referrer": "https://google.com/search",
    "properties": {
      "page_title": "User Dashboard",
      "load_time_ms": 1250,
      "viewport_width": 1920,
      "viewport_height": 1080
    },
    "user_properties": {
      "account_type": "premium",
      "signup_date": "2023-06-15",
      "total_sessions": 45,
      "last_login": "2024-01-14T18:30:00Z"
    }
  },
  {
    "user_id": "user_456",
    "session_id": "sess_def456",
    "timestamp": "2024-01-15T10:12:45.567Z",
    "event_type": "button_click",
    "page": "/products",
    "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15",
    "ip_address": "10.0.0.50",
    "properties": {
      "button_text": "Add to Cart",
      "product_id": "PROD-789",
      "category": "Electronics",
      "price": 299.99
    },
    "user_properties": {
      "account_type": "free",
      "signup_date": "2024-01-10",
      "total_sessions": 3,
      "last_login": "2024-01-15T08:00:00Z"
    }
  },
  {
    "user_id": "user_123",
    "session_id": "sess_abc789",
    "timestamp": "2024-01-15T09:47:15.890Z",
    "event_type": "form_submission",
    "page": "/contact",
    "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36",
    "ip_address": "192.168.1.100",
    "properties": {
      "form_name": "contact_us",
      "fields_completed": 5,
      "time_to_complete_seconds": 180,
      "success": true
    },
    "user_properties": {
      "account_type": "premium",
      "signup_date": "2023-06-15",
      "total_sessions": 45,
      "last_login": "2024-01-14T18:30:00Z"
    }
  }
]