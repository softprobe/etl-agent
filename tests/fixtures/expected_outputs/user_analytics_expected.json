{
  "business_entities": [
    {
      "name": "User",
      "type": "entity",
      "primary_key": "user_id", 
      "description": "Platform user"
    },
    {
      "name": "Session",
      "type": "activity",
      "primary_key": "session_id",
      "description": "User session tracking"
    },
    {
      "name": "Event",
      "type": "behavioral",
      "primary_key": null,
      "description": "User behavior events"
    }
  ],
  "suggested_schema": {
    "user_events": {
      "user_id": {"type": "STRING", "nullable": false, "description": "User identifier"},
      "session_id": {"type": "STRING", "nullable": false, "description": "Session identifier"},
      "timestamp": {"type": "TIMESTAMP", "nullable": false, "description": "Event timestamp"},
      "event_type": {"type": "STRING", "nullable": false, "description": "Type of event"},
      "page": {"type": "STRING", "nullable": true, "description": "Page where event occurred"},
      "user_agent": {"type": "STRING", "nullable": true, "description": "User agent string"},
      "ip_address": {"type": "STRING", "nullable": true, "description": "User IP address"},
      "referrer": {"type": "STRING", "nullable": true, "description": "Referrer URL"}
    },
    "event_properties": {
      "user_id": {"type": "STRING", "nullable": false, "description": "User identifier"},
      "session_id": {"type": "STRING", "nullable": false, "description": "Session identifier"},
      "timestamp": {"type": "TIMESTAMP", "nullable": false, "description": "Event timestamp"},
      "property_key": {"type": "STRING", "nullable": false, "description": "Property name"},
      "property_value": {"type": "STRING", "nullable": true, "description": "Property value"},
      "property_type": {"type": "STRING", "nullable": true, "description": "Data type of property"}
    },
    "user_attributes": {
      "user_id": {"type": "STRING", "nullable": false, "description": "User identifier"},
      "account_type": {"type": "STRING", "nullable": true, "description": "User account type"},
      "signup_date": {"type": "DATE", "nullable": true, "description": "User signup date"},
      "total_sessions": {"type": "INTEGER", "nullable": true, "description": "Total session count"},
      "last_login": {"type": "TIMESTAMP", "nullable": true, "description": "Last login timestamp"}
    }
  },
  "confidence_scores": {
    "schema_detection": 0.88,
    "business_entity_detection": 0.85,
    "data_type_inference": 0.90,
    "relationship_detection": 0.80
  },
  "potential_issues": [
    {
      "type": "dynamic_properties",
      "description": "Event properties have dynamic keys requiring normalization",
      "severity": "high",
      "suggested_solution": "Create separate event_properties table with key-value pairs"
    },
    {
      "type": "nested_objects", 
      "description": "Properties and user_properties are nested objects",
      "severity": "medium",
      "suggested_solution": "Flatten or normalize into separate tables"
    }
  ]
}